### DISCUSSION TITLE
Example Timer Model

### DISCUSSION THREAD

**Discussion by gordon1213** [dscripka/openWakeWord#185] [Q&A] [answered]
Hello! I'm new to ML and I'm interested in making something similar to how the timer model is able to identify phrases to different binary detection groups. I'm aware of the ability to have multiple phrases respond to the same value as explained in the custom_model yml, but how is the timer able to identify different phrases? Would combining the onnx/tflite files achieve this? Thanks!

**Comment by dscripka** [comment, marked as answer]
When I was first building openWakeWord, I did experiment with multiple possible labels/classes, and the timer model was an example of that. It worked reasonably well, but in my testing since then I've noticed a few issues:

1) Training these models is more complex, and it's more difficult to balance between low false-positives and high classification accuracy between the different classes. Unfortunately, they can't be created by just combining the files for other models, and they have to be trained fully from scratch.

2) Increasing the number of classes decreases accuracy significantly, especially past ~5 classes.

I think the concept is still worthwhile, but I haven't yet had time to adjust the `training.py` code to fully support these types of multiclass models.

However, I have recently released another project, [openSpeechToIntent](https://github.com/dscripka/openSpeechToIntent) that performs a very similar task with (in my view) a much more effective and easy-to-use approach. This might be a good alternative, until openWakeWord can fully support these types of models.

**Reply by gordon1213** [reply]
I'll look into the openSpeechToIntent Library. Thanks for the suggestion!
